haskell/src/Stablecoin/Client/Cleveland/IO.hs

```diff
@@ -108,22 +109,25 @@ naturalParser = decimal
 textParser :: Parser Text
 textParser = fromString <$> some printChar
 
+contractAddressParser :: Parser ContractAddress
+contractAddressParser = mkAddressParser parseKindedAddress
```

Comment:
Perhaps this function (and maybe `mkAddressParser` itself too) is worth upstreaming as well

Reply:
Sure, but I think that seems a bit odd, `Morley.Michelson.Parser.Value` doesn't export any megaparsec parser for any typed michelson type (`Address`, `Mutez`, `MText`, etc) :thinking: We only have parsers for untyped values (e.g. `stringLiteral`), which we typecheck later.

Should we add megaparsec parsers for all of these?

Reply:
Personally, I don't think we should add something like this to morley, because the exact parser implementation depends on the context.

If you're parsing an address separated by spaces/periods (e.g. if the input string is `My address is tz1etc.`), then the parser should probably consume all alphanumeric characters, and then feed them into `Morley.Tezos.Address.parseAddress`.

But if you're parsing an address delimited by quotes (e.g. `PUSH address "tz1etc"`) then you probably want the parser to consumer all characters until it finds a `"`, and then feed them into `Morley.Tezos.Address.parseAddress`.

There is no generic one size fits all solution :/

Reply:
mmmmmh, yeah, you are right, those are fair points :ok_hand: 

TBH I think that for a moment I was thinking of options parsers for CLI with `optparse-applicative` and got confused here.

Those we do cover more extensively (which makes sense to me) and we already have parsers for different address types there, so no need to add more :ok_hand: